ecmascript/parser/src/parser/expr/ops.rs

```diff
@@ -322,7 +322,11 @@ impl<'a, I: Tokens> Parser<I> {
             })));
         }
 
-        if (self.ctx().in_async || self.syntax().top_level_await()) && is!(self, "await") {
+        if is!(self, "await") {
+            let ctx = self.ctx();
+            if ctx.in_function && !ctx.in_async {
+                syntax_error!(self, SyntaxError::AwaitInFunction);
```

Comment:
I found that this error can be recovered, so we can report more issue in a single file.